themes/bootstrap3/js/check_save_statuses.js

```diff
@@ -126,10 +141,8 @@ function checkSaveStatuses(_container) {
       }, $(ajaxItems[i]));
     }
   }
-  // Stop looking for a scroll loader
-  if (saveStatusObserver) {
-    saveStatusObserver.disconnect();
-  }
+
+  VuFind.emit("save-status-done");
```

Comment:
Same question for this one -- do we need it? Can't we rely on the queue emitting the status when it's done processing?

Reply:
@crhallberg, a time limit seems like a reasonable compromise, though I think it's inevitable that this will result in incomplete results in some situations (e.g. printing a results page with 100 items, using an ILS that runs slowly). Maybe the time limit should be relative to the last activity, rather than to the start of the loading process? And would it be more palatable if we had a "Loading..." message up while things get prepared for printing?

(Note that my goal is to do this as simply as possible, so don't let any of my ideas cause excessive complication... just brainstorming here!)

Reply:
I believe I have covered all cases but maybe a timeout as a failsafe would be good to add.

PS. I cannot wait to rewrite all of this logic for the new theme.